modules/3d/src/ptcloud/ptcloud_utils.hpp

```diff
@@ -73,6 +73,92 @@ inline void _getMatFromInputArray(InputArray input_pts, Mat &mat,
 
 }
 
+/** @brief Copy the data xyz of the point by specifying the indexs.
+ *
+ * @param src CV_32F Mat with size Nx3/3xN.
+ * @param[out] dst CV_32F Mat with size Mx3/3xM.
+ * @param idxs The index of the point copied from src to dst.
+ * @param dst_size The first dst_size of idxs is valid.
+ *                 If it is less than 0 or greater than idxs.size(),
+ *                 it will be automatically adjusted to idxs.size().
+ * @param arrangement_of_points The arrangement of point data in the matrix, \n
+ *                              0 by row (Nx3, [x1, y1, z1, ..., xn, yn, zn]),  \n
+ *                              1 by column (3xN, [x1, ..., xn, y1, ..., yn, z1, ..., zn]).
+ */
+inline void
+copyPointDataByIdxs(const Mat &src, Mat &dst, const std::vector<int> &idxs, int dst_size = -1,
```

Comment:
I think this makes more sense to put these functions into .cpp files.
1. This function does not look like it needs to be inlined.
2. It will be handy to parallelize it.

Reply:
Modified.